yt/frontends/ramses/data_structures.py

```diff
@@ -197,13 +212,101 @@ def fill(self, fd, fields, selector, file_handler):
         # Initializing data container
         for field in fields:
             tr[field] = np.zeros(cell_count, 'float64')
-
         fill_hydro(fd, file_handler.offset,
-                   file_handler.level_count, levels, cell_inds,
+                   file_handler.level_count,
+                   [self.domain_id-1],
+                   levels, cell_inds,
                    file_inds, ndim, all_fields, fields, tr,
                    oct_handler)
         return tr
 
+    def _fill_with_ghostzones(self, fd, fields, selector, file_handler, num_ghost_zones):
+        ndim = self.ds.dimensionality
+        ncpu = self.ds.parameters['ncpu']
+        # Here we get a copy of the file, which we skip through and read the
+        # bits we want.
+        oct_handler = self.oct_handler
+        all_fields = [f for ft, f in file_handler.field_list]
+        fields = [f for ft, f in fields]
+        tr = {}
+
+        cell_count = selector.count_octs(self.oct_handler, self.domain_id) * self.nz**ndim
+
+        levels, cell_inds, file_inds, domains = self.oct_handler.file_index_octs_with_ghost_zones(
+            selector, self.domain_id, cell_count)
+
+        # Initializing data container
+        for field in fields:
+            tr[field] = np.zeros(cell_count, 'float64')
+        fill_hydro(fd, file_handler.offset,
+                   file_handler.level_count,
+                   list(range(ncpu)),
+                   levels, cell_inds,
+                   file_inds, ndim, all_fields, fields, tr,
+                   oct_handler,
+                   domains=domains)
+        return tr
+
+    @property
+    def fwidth(self):
+        fwidth = super(RAMSESDomainSubset, self).fwidth
+        if self._num_ghost_zones > 0:
+            fwidth = super(RAMSESDomainSubset, self).fwidth.reshape(-1, 8, 3)
+            n_oct = fwidth.shape[0]
+            new_fwidth = np.zeros((n_oct, self.nz**3, 3), dtype=fwidth.dtype)
+            new_fwidth[:, :, :] = fwidth[:, 0:1, :]
+            fwidth = new_fwidth.reshape(-1, 3)
+        return fwidth
+
+    @property
+    def fcoords(self):
+        fcoords = super(RAMSESDomainSubset, self).fcoords
+        num_ghost_zones = self._num_ghost_zones
+        if num_ghost_zones == 0:
+            return fcoords
```

Comment:
`fcoords` AFAICT is only used if `self._num_ghost_zones == 0`. Wouldn't it make sense to move it inside the if clause?

```
if self._num_ghost_zones == 0:
    return super(RAMSESDomainSubset, self).fcoords
```

I'm not sure how costly it is and if it really matters...